crates/primitives/src/transaction/access_list.rs

```diff
@@ -3,6 +3,19 @@
 /// Re-export from `alloy_eips`.
 #[doc(inline)]
 pub use alloy_eips::eip2930::{AccessList, AccessListItem};
+use revm_primitives::U256;
+use serde::{Deserialize, Serialize};
+
+/// `AccessListWithGasUsedAndError` for handling error from `eth_createAccessList`
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AccessListWithGasUsedAndError {
+    /// List with accounts accessed during transaction.
+    pub access_list: AccessList,
+    /// Estimated gas used with access list.
+    pub gas_used: U256,
+    /// Optional error message if the transaction failed.
+    pub error: Option<String>,
```

Comment:
this should be skipped if none

 #[serde(default, skip_serializing_if = "Option::is_none")]